src/fielddef.ts

```diff
@@ -173,7 +173,18 @@ export interface TextFieldDef<F> extends FieldDef<F> {
 
 export type ConditionalTextDef<F> = Conditional<TextFieldDef<F>, ValueDef<string|number|boolean>>;
 
-export type ChannelDef<F> = FieldDef<F> | ValueDef<any>;
+export type ChannelDef<F> = Conditional<FieldDef<F>, ValueDef<any>>;
+
+export function isConditionalDef<F>(channelDef: ChannelDef<F>): channelDef is Conditional<FieldDef<F>, ValueDef<any>> {
+  return !!channelDef && !!channelDef.condition;
+}
+
+/**
+ * Return if a channelDef is a ConditionalValueDef with ConditionFieldDef
+ */
+export function hasConditionFieldDef<F>(channelDef: ChannelDef<F>): channelDef is (ValueDef<any> & {condition: FieldDef<F>}) {
+  return !!channelDef && !!channelDef.condition && isFieldDef(channelDef.condition);
+}
 
 export function isFieldDef<F>(channelDef: ChannelDef<F>): channelDef is FieldDef<F> | PositionFieldDef<F> | LegendFieldDef<F, any> | OrderFieldDef<F> | TextFieldDef<F> {
   return !!channelDef && (!!channelDef['field'] || channelDef['aggregate'] === 'count');
```

Comment:
Yeah,  say a channelDef could be 

1) FieldDef (with no condition)
2) FieldDef with ConditionValueDef

(1 & 2 are `ConditionalFieldDef` as I make `condition?` so I don't have to declare too many types)

3) ValueDef (with no condition)
4) ValueDef with ConditionFieldDef
5) ValueDef with ConditionValueDef

(3-5 are `ConditionalValueDef` as I make `condition?` so I don't have to declare too many types)

6) ConditionFieldDef only  (akin to if without else)
7) ConditionValueDef only 

(6-7 are `ConditionOnlyDef`)

`isFieldDef` is more scoped to one object and can check if the object is a fieldDef.  
The input can be **either**:
- a)  the parent object of a channelDef (`encoding.<channel>`), which will return `true` for 1 or 2, 
- b) the condition object of a channelDef  (`encoding.<channel>.condition`), which will return true for case 4 or 6.  

On the other hand, `getFieldDef` is for getting the `fieldDef` from a channelDef. 
Given a channelDef (from `encoding.<channel>`), this will return either the parent channelDef object (for case 1 or 2) or its condition object (for case 4 or 6).  

In a way, getFieldDef is more like `getFieldDefFromChannelDef` but it is kinda verbose, so I just chose the brief getFieldDef. That said, I agree that it can be confusing. 

If you have better name suggestions, let me know :)